src/gtimelog/ui/components/windows.py

```diff
@@ -0,0 +1,767 @@
+import datetime
+import functools
+import os
+import smtplib
+from contextlib import closing
+from email.utils import parseaddr
+from gettext import gettext as _
+from gi.repository import Gtk, Gdk, GLib, Gio, GObject, Soup
+
+from gtimelog import root_logger, DEBUG
+from gtimelog.core.exceptions import EmailError
+from gtimelog.ui.components.services import start_smtp_password_lookup, Authenticator
+from gtimelog.core.settings import Settings
+from gtimelog.core.timelog import TaskList
+from gtimelog.core.time import TimeLog
+from gtimelog.core.utils import mark_time, prev_month, prepare_message, virtual_day, next_month
+from gtimelog.paths import UI_FILE, MENUS_UI_FILE
+from gtimelog.ui.components.entries import TaskEntry
+from gtimelog.ui.components.utils import copy_properties, swap_widget, internationalised_format_duration, \
+    MAIL_PROTOCOLS, REPORT_KINDS
```

Comment:
I should mention that I prefer

```python
from gtimelog.ui.components.utils import (
    copy_properties, ...,
   ...,
)
```

for wrapping long import statements.

Reply:
Applied